src/Tickets/REST/TEC/V1/Endpoints/Ticket.php

```diff
@@ -231,6 +233,84 @@ public function update_schema(): OpenAPI_Schema {
 		return $schema;
 	}
 
+	/**
+	 * Filters ticket stock, capacity, and mode parameters according to business logic.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $params The request parameters.
+	 *
+	 * @return array The filtered parameters.
+	 */
+	protected function filter_create_params( array $params ): array {
+		// Normalize natural-language date strings (e.g., "now", "next friday 6pm").
+		foreach ( [ 'start_date', 'end_date', 'sale_price_start_date', 'sale_price_end_date' ] as $date_key ) {
+			if ( isset( $params[ $date_key ] ) && is_string( $params[ $date_key ] ) ) {
+				$parsed = $this->normalize_date_text_to_mysql( $params[ $date_key ] );
+				if ( null !== $parsed ) {
+					$params[ $date_key ] = $parsed;
+				}
+			}
```

Comment:
This should be handled either on the MCP level or on the repository level.

Reply:
Moved to the repository.